src/Corax/IndexSearcher/IndexSearcher.cs

```diff
@@ -264,6 +269,12 @@ private ByteStringContext<ByteStringMemoryCache>.InternalScope AnalyzeTerm(Analy
     public AllEntriesMatch AllEntries() => new AllEntriesMatch(this, _transaction);
    public TermMatch EmptyMatch() => TermMatch.CreateEmpty(this, Allocator);
 
+   public long GetDictionaryIdFor(Slice field)
+   {
+       var terms = _fieldsTree?.CompactTreeFor(field);
+       return terms?.DictionaryId ?? -1;
```

Comment:
I tried to write a test that will trigger `-1` but it seems there is a problem when not all documents have sorted field.

Tests (SortingMatch throws NRE):
https://gist.github.com/maciejaszyk/e3d32d7cf3f87e09089c556ed912466c

Reply:
@ayende please make sure the failing tests found by Maciej will be fixed and added to the PR as well

Reply:
Looking into this

Reply:
Fixed